### PR TITLE
add tools/anchor_cluster.py

### PR DESCRIPTION
**Pull request by wangxinxin08**
add tools/anchor_cluster.py

fix bugs

add docstring

### CODE REVIEW

File: tools/anchor_cluster.py [hunk 1]
```diff
@@ -0,0 +1,358 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
```
Comment (heavengate) [inline tools/anchor_cluster.py:1]: 2019 -> 2020

File: tools/anchor_cluster.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+            cache (bool): whether using cache
+            iters (int): kmeans algorithm iters
+            verbose (bool): whether print results
+        """
+        self.dataset = dataset
+        self.size = size
+        self.iters = iters
+
+    def print_result(self, centers):
+        print('%d anchor cluster result: [w, h]' % self.n)
```
Comment (heavengate) [inline tools/anchor_cluster.py:128]: 用logger吧，跟train/eval这些一致

File: tools/anchor_cluster.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+    # check if set use_gpu=True in paddlepaddle cpu version
+    check_gpu(cfg.use_gpu)
+    # check if paddlepaddle version is satisfied
+    check_version()
+
+    # get dataset
+    dataset = cfg['TrainReader']['dataset']
+    if FLAGS.size:
+        if ',' in FLAGS.size:
+            size = list(map(int, FLAGS.size.split(',')))
```
Comment (heavengate) [inline tools/anchor_cluster.py:334]: check一下len(size) == 2?

**Review by heavengate — COMMENTED** [review]
1. 可以贴一下运行日志和COCO上结果
2. yolo系列的文档里面加一下这个脚本的使用方法

**Comment by wangxinxin08** [comment]
> 1. 可以贴一下运行日志和COCO上结果
> 2. yolo系列的文档里面加一下这个脚本的使用方法
在coco上的运行日志和结果：
loading annotations into memory...
Done (t=20.30s)
creating index...
index created!
100%███████████████████████████████████████████████████████| 118287/118287 [06:38<00:00, 297.18it/s]
2020-09-17 05:11:34,037-WARNING: Extremely small objects found. 3880 of 849949labels are < 3 pixels in width or height
2020-09-17 05:11:34,062-INFO: Running kmeans for 9 anchors on 849904 points...
Evolving anchors with Genetic Algorithm: fitness = 0.6893: 100%███████████████████████████████████████████████████████| 1000/1000 [10:18<00:00, 1.62it/s]
2020-09-17 05:22:17,076-INFO: thresh=0.25: 0.9956 best possible recall, 4.35 anchors past thr
2020-09-17 05:22:17,160-INFO: n=9, img_size=[640, 640], metric_all=0.291/0.690-mean/best, past_thresh=0.469-mean:
2020-09-17 05:22:17,160-INFO: 9 anchor cluster result: [w, h]
2020-09-17 05:22:17,160-INFO: [10, 12]
2020-09-17 05:22:17,160-INFO: [15, 33]
2020-09-17 05:22:17,160-INFO: [34, 24]
2020-09-17 05:22:17,160-INFO: [31, 62]
2020-09-17 05:22:17,160-INFO: [75, 52]
2020-09-17 05:22:17,161-INFO: [68, 120]
2020-09-17 05:22:17,161-INFO: [160, 118]
2020-09-17 05:22:17,161-INFO: [143, 256]
2020-09-17 05:22:17,161-INFO: [369, 320]

**Review by heavengate — APPROVED** [review]

**Pull request merged**